### PR TITLE
#8 Setup proper heroku deployment

### PR DESCRIPTION
**Pull request by JustasBelevicius**
- Runs the app on `https://petbook-back.herokuapp.com/graphiql`
- Postgres db had to be initialised by hand, I might create an action later, but it would have to drop all tables before updating - loosing all data we had. For now, I believe it is better to have it be difficult to update, to avoid accidental data drops. Might want to eventually have some kind of `database/updates/1.sql` files to update the tables without loosing data.

### CODE REVIEW

**Comment by JustasBelevicius** [comment]
Note! We don't want to run the Deploy Heroku job on pull_request open :D

File: src/services/postgres.ts [hunk 1]
```diff
@@ -29,13 +30,23 @@ const pgClient = (pool: Pool): PostgresClient => ({
...
+        },
+      }
+    : {
+        user: process.env.POSTGRES_USER,
+        host: process.env.POSTGRES_HOST,
+        database: process.env.POSTGRES_DB,
+        password: process.env.POSTGRES_PASSWORD,
+        port: parseInt(process.env.POSTGRES_PORT || '5432', 10),
+      };
+  console.log(config);
```
Comment (psaulenas) [inline src/services/postgres.ts:48]: Could you please remove `console.log`?

File: package.json [hunk 2]
```diff
@@ -4,11 +4,13 @@
...
   "scripts": {
-    "start": "tsc && nodemon dist/index.js",
+    "start": "nodemon dist/index.js",
     "test": "echo \"Error: no test specified\" && exit 1",
     "debug": "tsc && nodemon --inspect-brk=0.0.0.0 dist/index.js",
     "check-lint": "eslint 'src/**/*.ts'",
-    "check-type": "tsc --noEmit --skipLibCheck"
+    "check-type": "tsc --noEmit --skipLibCheck",
+    "build": "tsc",
+    "start-prod": "npm run build && npm run start"
```
Comment (psaulenas) [inline package.json:13]: Why did you also create `start-prod`?

**Review by psaulenas — APPROVED** [review]
GJ! Left a few comments

File: src/services/postgres.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import console from 'console';
```
Comment (psaulenas) [inline src/services/postgres.ts:1]: Seems not needed import

Comment (JustasBelevicius) [reply src/services/postgres.ts:1]: Will add a rule to husky if possible

Comment (JustasBelevicius) [reply src/services/postgres.ts:1]: Looks like the current `eslint` setup checks for unused imports. This one is used in the PostgresClient disconnect callback, if an error is caught.

**Review by AzisK — APPROVED** [review]
Looks smooth, good job

File: src/services/postgres.ts (hunk 3 above)
Comment (psaulenas) [reply src/services/postgres.ts:1]: My guess would be that your IDE did auto-import. Removing this import shouldn't affect anything, you don't need to import `console`

**Review by ignasposka — APPROVED** [review]

File: src/services/postgres.ts (hunk 3 above)
Comment (JustasBelevicius) [reply src/services/postgres.ts:1]: That's exactly what happened :D

**Pull request merged**